src/psyclone/tests/psyir/transformations/arrayrange2loop_trans_test.py

```diff
@@ -40,445 +40,355 @@
+    assert ArrayRange2LoopTrans().name == "ArrayRange2LoopTrans"
 
-    '''
-    array_symbol = DataSymbol("x", ArrayType(REAL_TYPE, [10]))
-    symbol_table.add(array_symbol)
-    symbol_n = symbol_table.lookup("n")
-    array_x = ArrayReference.create(array_symbol,
-                                    [create_stepped_range(symbol_n)])
-    array_symbol = DataSymbol("z", ArrayType(REAL_TYPE, [10, 10]))
-    symbol_table.add(array_symbol)
-    array_z = ArrayReference.create(array_symbol,
-                                    [Literal("1", INTEGER_TYPE),
-                                     create_stepped_range(symbol_n)])
-    array_symbol = DataSymbol("a", ArrayType(REAL_TYPE, [10]))
-    array_a = ArrayReference.create(array_symbol, [Literal("1", INTEGER_TYPE)])
-    symbol_table.add(array_symbol)
-    mult = BinaryOperation.create(
-        BinaryOperation.Operator.MUL, array_x, array_z)
-    add = BinaryOperation.create(BinaryOperation.Operator.ADD, mult, array_a)
-    return add
-
-
-def test_transform():
-    '''Check that it is possible to create an instance of
-    ArrayRange2LoopTrans and that it is a Transformation.
 
-    '''
-    assert ArrayRange2LoopTrans()
-    assert isinstance(ArrayRange2LoopTrans(), Transformation)
-
-
-# The parameterised tests are 1: x(:)=0.0, 2: x(:)=y(n,:), 3:
-# y(n,:)=x(:), 4: y2(:,:)=z(:,n,:) and 5:
-# y3(n,2:n:2)=x(2:n:2)*z(1,2:n:2)+a(1)
-@pytest.mark.parametrize("lhs_create,rhs_create,expected",
-                         [(create_array_x, create_literal,
-                           "  do idx = LBOUND(x, dim=1), UBOUND(x, dim=1), 1\n"
-                           "    x(idx) = 0.0\n"),
-                          (create_array_x, create_array_y,
-                           "  do idx = LBOUND(x, dim=1), UBOUND(x, dim=1), 1\n"
-                           "    x(idx) = y(n,idx)\n"),
-                          (create_array_y, create_array_x,
-                           "  do idx = LBOUND(y, dim=2), UBOUND(y, dim=2), 1\n"
-                           "    y(n,idx) = x(idx)\n"),
-                          (create_array_y_2d_slice, create_array_z,
-                           "  do idx = LBOUND(y2, dim=2), UBOUND(y2, dim=2),"
-                           " 1\n"
-                           "    y2(:,idx) = z(:,n,idx)\n"),
-                          (create_array_y_slice_subset, create_expr,
-                           "  do idx = 2, n, 2\n"
-                           "    y3(n,idx) = x(idx) * z(1,idx) + a(1)")])
-def test_transform_apply(lhs_create, rhs_create, expected, tmpdir):
+@pytest.mark.parametrize(
+        "code, expected",
+
+        # Scalar RHS
+        [("integer, dimension(:) :: x\n"
+          "x(:) = 0.0",
+          "  do idx = LBOUND(x, dim=1), UBOUND(x, dim=1), 1\n"
+          "    x(idx) = 0.0\n"),
+
+         # Array LHS and RHS
+         ("integer, dimension(:) :: x, y\n"
+          "x(:) = y(:)\n",
+          "  do idx = LBOUND(x, dim=1), UBOUND(x, dim=1), 1\n"
```

Comment:
I believe the code is right, and psyclone does the right think. In fact, if you add
```
write(*,*) LBOUND(a1)
write(*,*) LBOUND(a2)
```
inside my_sub it will print 1 and 1